### PR TITLE
actions, publish: Push the main core image

### PR DESCRIPTION
**Pull request by qinqon**
After a PR is merge the kiagnose core image has to be pushed so users
can consume it. This change push the core kiagnose image on PR merges at
quay.io/kiagnose/kiagnose:main.

Notes for reviewers:
This cannot be tested on the PR since secrets are not passed to jobs running at PRs.
Tested with [act](https://github.com/nektos/act)

### CODE REVIEW

File: automation/make.sh [hunk 1]
```diff
@@ -22,6 +22,8 @@ set -e
 CRI=${CRI:-podman}
 
 CORE_BINARY_NAME="kiagnose"
+CORE_IMAGE_REPO=${CORE_IMAGE_REPO:-quay.io}
```
Comment (EdDev) [inline automation/make.sh:25]: Isn’t it a registry? The name hints it is a repo.

File: automation/make.sh [hunk 2]
```diff
@@ -45,9 +46,12 @@ while true; do
     --build-core-image)
         OPT_BUILD_CORE_IMAGE=1
         ;;
+    --push-core-image)
+        OPT_PUSH_CORE_IMAGE=1
```
Comment (EdDev) [inline automation/make.sh:50]: It needs to be added to the defaults, but not triggered automatically.

**Review by EdDev — COMMENTED** [review]
This is really nice.
Some point inline…

**Comment by EdDev** [comment]
This also needs to be limited/filtered per changes in the cmd and kiagnose folders only. If possible, please excluding tests files will also be nice (but not a must).

**Comment by qinqon** [comment]
> This also needs to be limited/filtered per changes in the cmd and kiagnose folders only. If possible, excluding tests files will also be nice (but not a must).

Not really needed, the time it takes does not affect PRs CI and the image sha is going to be the same so pushing will not overwrite anything.

**Comment by EdDev** [comment]
> Not really needed, the time it takes does not affect PRs CI and the image sha is going to be the same so pushing will not overwrite anything, is better to just delegate container hash comparation on pushing.

Are you sure the build artifact is not affected? Remember that there is no cache between the runs, unlike when run locally.

Anyway, we can take it in and examine as it goes.

**Review by EdDev — APPROVED** [review]
Thank you!

**Pull request merged**